sympy/solvers/tests/test_solveset.py

```diff
@@ -178,6 +178,18 @@ def test_issue_11536():
     assert solveset(0**x - 1, x, S.Reals) == FiniteSet(0)
 
 
+def test_issue_17479():
+    import sympy as sb
+    from sympy.solvers.solveset import nonlinsolve
+    x, y, z = sb.symbols("x, y, z")
+    f = (x**2 + y**2)**2 + (x**2 + z**2)**2 - 2*(2*x**2 + y**2 + z**2)
+    fx = sb.diff(f, x)
+    fy = sb.diff(f, y)
+    fz = sb.diff(f, z)
+    sol = nonlinsolve([fx, fy, fz], [x, y, z])
+    assert len(sol) == 18
```

Comment:
Are we sure that there should be 18 solutions here. In a PR I'm writing I'm getting 20 solutions. Mapping the solutions through expand and eliminating duplicates I get 16 either way.

I just want to ask here since this is a recently added test. Is the solution given on master absolutely correct or is this just testing that an exception doesn't get raised?

Reply:
> Are we sure that there should be 18 solutions here. In a PR I'm writing I'm getting 20 solutions. Mapping the solutions through expand and eliminating duplicates I get 16 either way.
> 
> I just want to ask here since this is a recently added test. Is the solution given on master absolutely correct or is this just testing that an exception doesn't get raised?

Sir I confirmed the solution from @jksuom, since I am new, I am not very sure, it would be nice if u confirm this from him.

Reply:
There are probably many ways of representing the solutions. This test only says that nonlinsolve currently returns 18 parametrized solutions that are not necessarily independent.

Reply:
Sir since I am new, can any member of the organisation help me understand the code base, or give me any idea as to how to understand it, because at the moment I am able to understand it in bits and pieces.

@jksuom @oscarbenjamin @smichr @sylee957